Fix use-after-free during ``Py_EndInterpreter``.
